### PR TITLE
Eco forms

### PR DESCRIPTION
**Pull request by casey1173**
Adds htmx eco input forms for design, cause, effect, and result.
- Includes a new NestedTerm for cause/effect
- Adds eco as a study type
- Adds tests for the eco forms

### CODE REVIEW

File: hawc/apps/eco/models.py [hunk 1]
```diff
@@ -360,9 +410,21 @@ class Meta:
...
         if self.measure_type:
-            return f"{self.study} | {self.cause.term.value} |{self.effect.term.value} | {self.measure_type.value}"
+            return f"{self.cause.study} | {self.cause.term.name} |{self.effect.term.name} | {self.measure_type.value}"
         else:
-            return f"{self.study} | {self.cause.term.value} | {self.effect.term.value}"
+            return f"{self.cause.study} | {self.cause.term.name} | {self.effect.term.name}"
+
+    def clone(self):
+        self.id = None
+        self.sort_order = self.sort_order + 1
```
Comment (casey1173) [inline hawc/apps/eco/models.py:419]: I went rogue here!!

File: hawc/apps/eco/lookups.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
+from selectable.base import ModelLookup
+from selectable.registry import registry
+
+from ..common.lookups import RelatedDistinctStringLookup
+from ..epi.models import Country
+from . import constants, models
+
+
+class RelatedCauseLookup(RelatedDistinctStringLookup):
```
Comment (shapiromatron) [inline hawc/apps/eco/lookups.py:9]: update these to the new [`dal` ](https://github.com/shapiromatron/hawc/pull/667) autocompletes

File: hawc/apps/eco/migrations/0003_auto_20220719_1256.py [hunk 3]
```diff
@@ -0,0 +1,193 @@
+# Generated by Django 3.2.12 on 2022-07-19 16:56
```
Comment (shapiromatron) [inline hawc/apps/eco/migrations/0003_auto_20220719_1256.py:1]: squash eco migrations? i think there just needs to be 2 migrations - one that creates all the models, and one that loads the data

File: hawc/apps/eco/models.py [hunk 4]
```diff
@@ -15,9 +18,21 @@ def __str__(self):
         return self.name
 
 
+class NestedTerm(MP_Node):
+    type_choices = (("CE", "cause/effect"),)
```
Comment (shapiromatron) [inline hawc/apps/eco/models.py:22]: move this to eco.constants as a models.TextChoices

File: hawc/apps/eco/views.py [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+    parent_template_name = "study"
+    model = models.Design
+    form_class = forms.DesignForm
+    template_name = "eco/design_update.html"
+
+    def get_context_data(self, **kwargs):
+        design_id = self.kwargs.pop("pk")
+        context = super().get_context_data(**kwargs)
+        study_id = self.object.study_id
+        context["causes"] = models.Cause.objects.filter(study=study_id)
```
Comment (shapiromatron) [inline hawc/apps/eco/views.py:38]: We can remove these 3 context items for update, detail, and delete. Instead, update the `get_queryset` method and just prefetch_related for these models; and then we can use the attributes directly on the instance

File: hawc/apps/eco/forms.py [hunk 6]
```diff
@@ -1,48 +1,147 @@
...
+    def __init__(self, *args, **kwargs):
+        study = kwargs.pop("parent", None)
+        super().__init__(*args, **kwargs)
+        if study:
+            self.instance.study = study
+
+    @property
+    def helper(self):
+        for fld in ("habitat_as_reported", "climate_as_reported"):
+            self.fields[fld].widget.attrs["class"] = "html5text"
```
Comment (shapiromatron) [inline hawc/apps/eco/forms.py:31]: remove the quill attributes from this; these can just be plain text, not quill editors

File: hawc/apps/eco/forms.py [hunk 7]
```diff
@@ -1,48 +1,147 @@
...
+    def __init__(self, *args, **kwargs):
+        design = kwargs.pop("parent", None)
+        super().__init__(*args, **kwargs)
+        if design:
+            self.instance.study = design.study
+
+    @property
+    def helper(self):
+        for fld in ("comment", "as_reported"):
+            self.fields[fld].widget.attrs["class"] = "html5text"
```
Comment (shapiromatron) [inline hawc/apps/eco/forms.py:74]: same here and elsewhere in the eco app, no html5text of wysiwyg editing (consistent w/ epiv2)

**Review by shapiromatron — COMMENTED** [review]
Looking great! a few edits, mostly minor

File: hawc/apps/eco/models.py (hunk 1 above)
Comment (shapiromatron) [reply hawc/apps/eco/models.py:419]: I'll dig into this a bit later, thanks!

**Review by shapiromatron — APPROVED** [review]

**Pull request merged**